querydsl-sql/src/main/java/com/querydsl/sql/dml/AbstractSQLClause.java

```diff
@@ -220,6 +237,13 @@ protected void reset() {
         cleanupMDC();
     }
 
+    protected Connection connection() {
+        if (conn == null && connProvider != null) {
+            conn = connProvider.get();
```

Comment:
Then put an assert here?
Just so there is an exception, that indicates that it's deliberate.